third-party/realdds/src/dds-device.cpp

```diff
@@ -19,11 +19,91 @@ std::mutex devices_mutex;
 }  // namespace
 
 
+dds_device::dds_stream::dds_stream( rs2_stream type, std::string group_name )
+    : _impl(std::make_shared< dds_stream::impl >( type, group_name ))
+{
+}
+
+void dds_device::dds_stream::add_video_profile( const rs2_video_stream & profile, bool default_profile )
+{
+    if ( _impl->_type != profile.type )
+    {
+        throw std::runtime_error( "profile of different type then stream" );
+    }
+
+    _impl->_video_profiles.push_back( { profile, default_profile } );
+}
+
+void dds_device::dds_stream::add_motion_profile( const rs2_motion_stream & profile, bool default_profile )
+{
+    if ( _impl->_type != profile.type )
+    {
+        throw std::runtime_error( "profile of different type then stream" );
+    }
+
+    _impl->_motion_profiles.push_back( { profile, default_profile } );
+}
+
+size_t dds_device::dds_stream::foreach_video_profile( std::function< void( const rs2_video_stream & profile, bool def_prof ) > fn ) const
+{
+    for ( auto profile : _impl->_video_profiles )
+    {
+        fn( profile.first, profile.second );
+    }
+
+    return _impl->_video_profiles.size();
+}
+
+size_t dds_device::dds_stream::foreach_motion_profile( std::function< void( const rs2_motion_stream & profile, bool def_prof ) > fn ) const
+{
+    for ( auto profile : _impl->_motion_profiles )
+    {
+        fn( profile.first, profile.second );
+    }
+
+    return _impl->_motion_profiles.size();
+}
+
+//std::string dds_device::dds_stream::get_name()
+//{
+//    std::stringstream ss;
+//    ss << rs2_stream_to_string( _impl->_type );
+//
+//    //Add index number to the name
+//    if ( _impl->_type == RS2_STREAM_DEPTH || _impl->_type == RS2_STREAM_COLOR || _impl->_type == RS2_STREAM_INFRARED )
+//    {
+//        if ( ! _impl->_video_profiles.empty() && _impl->_video_profiles[0].first.index != 0 )
+//        {
+//            ss << " " << _impl->_video_profiles[0].first.index;
+//        }
+//    }
+//    else if ( _impl->_type == RS2_STREAM_GYRO || _impl->_type == RS2_STREAM_ACCEL )
+//    {
+//        if ( ! _impl->_motion_profiles.empty() && _impl->_motion_profiles[0].first.index != 0 )
+//        {
+//            ss << " " << _impl->_motion_profiles[0].first.index;
+//        }
+//    }
+//    else
+//    {
+//        throw std::runtime_error( "trying to get name of unsupported stream type " + std::to_string( _impl->_type ) );
+//    }
+//
+//    return ss.str();
+//}
+
 std::shared_ptr< dds_device > dds_device::find( dds_guid const & guid )
+{
+    return find_internal( guid, true );
+}
+
+std::shared_ptr< dds_device > dds_device::find_internal( dds_guid const & guid, bool shouldLock )
```

Comment:
Why `should_lock`?
Just remove it and lock outside, always... (like `create` does)

Reply:
Done